src/DynamoDbQueryBuilder.php

```diff
@@ -315,15 +327,26 @@ protected function conditionsContainKey()
         return $conditionValue;
     }
 
+    /**
+     * Check if conditions "where" contain index key
+     * For composite index key, it will return false if the conditions don't have all composite key.
+     *
+     * @return array|bool false or array ['name' => 'index_name', 'keys' => ['hash' => 'hash_key', 'range' => 'range_key']]
+     */
     protected function conditionsContainIndexKey()
     {
         if (empty($this->where)) {
             return false;
         }
 
-        foreach ($this->model->getDynamoDbIndexKeys() as $key => $name) {
-            if (isset($this->where[$key])) {
-                return $key;
+        foreach ($this->model->getDynamoDbIndexKeys() as $name => $keyArr) {
+            $conditionKeys = array_keys($this->where);
+            $keys = array_values($keyArr);
+            if (count(array_intersect($conditionKeys, $keys)) === count($keys)) {
+                return [
+                    'name' => $name,
+                    'keys' => $keyArr
```

Comment:
Do you think we can call it `'keysInfo'` or something so it's a bit clearer? Since we have `$keys` being an array of strings already, it could be confusing.